mbedtls/src/error.rs

```diff
@@ -95,59 +147,79 @@ impl From<Infallible> for Error {
     }
 }
 
+impl From<LoError> for Error {
+    fn from(error: LoError) -> Error {
+        Error::LowLevel(error)
+    }
+}
+
+impl From<HiError> for Error {
+    fn from(error: HiError) -> Error {
+        Error::HighLevel(error)
+    }
+}
+
+impl BitOr<LoError> for HiError {
+    type Output = Error;
+    fn bitor(self, rhs: LoError) -> Self::Output {
+        Error::HighAndLowLevel(self, rhs)
+    }
+}
+impl BitOr<HiError> for LoError {
+    type Output = Error;
+    fn bitor(self, rhs: HiError) -> Self::Output {
+        Error::HighAndLowLevel(rhs, self)
+    }
+}
+
+impl From<c_int> for Error {
+    fn from(x: c_int) -> Error {
+        let (high_level_code, low_level_code) = (-x & HiError::mask(), -x & LoError::mask());
+        if -x & (HiError::mask() | LoError::mask()) != -x || x >= 0 {
+            return Error::Other(x);
+        }
+        else if high_level_code == 0 {
```

Comment:
Nit: non-standard formatting (same for all the other branches below)
```suggestion
        } else if high_level_code == 0 {
```